homeassistant/components/recorder/core.py

```diff
@@ -619,6 +620,13 @@ def _wait_startup_or_shutdown(self) -> object | None:
             return SHUTDOWN_TASK
 
     def run(self) -> None:
+        """Run the recorder thread."""
+        try:
+            self._run()
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected error in recorder thread")
```

Comment:
If the recorder thread collapsed due to an exception it would silently die. I screwed up a test refactor when I was working on this and it took me a while to figure it out

Reply:
The threading.excepthook  will catch this in production. Its only in tests that it doesn't happen

Reply:
removed it